src/Uno.UWP/Data/Pdf/PdfDocument.Android.cs

```diff
@@ -0,0 +1,133 @@
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Android.Graphics.Pdf;
+using Android.OS;
+
+namespace Windows.Data.Pdf;
+
+public partial class PdfDocument : IDisposable
+{
+	private readonly PdfRenderer pdfRenderer;
+
+	private PdfDocument(PdfRenderer pdfRenderer)
+	{
+		this.pdfRenderer = pdfRenderer ?? throw new ArgumentNullException(nameof(pdfRenderer));
+	}
+
+	public bool IsPasswordProtected => false;
+
+	public uint PageCount => (uint)pdfRenderer.PageCount;
+
+	public PdfPage GetPage(uint pageIndex)
+	{
+		if (pageIndex >= pdfRenderer.PageCount)
+		{
+			throw new ArgumentOutOfRangeException(nameof(pageIndex), $"In this document the page index cannot be {pageIndex}. Page count is {pdfRenderer.PageCount}");
+		}
+
+		var pdfPage = pdfRenderer.OpenPage((int)pageIndex);
+		return new PdfPage(pdfPage);
+	}
+
+	public static Foundation.IAsyncOperation<PdfDocument> LoadFromFileAsync(Storage.IStorageFile file)
+	{
+		return LoadFromFileAsync(file, null);
+	}
+
+	public static Foundation.IAsyncOperation<PdfDocument> LoadFromFileAsync(Storage.IStorageFile file, string password)
+	{
+		if (!string.IsNullOrWhiteSpace(password))
+		{
+			// password protected PDF files are not supported by PdfRenderer in Android
+			throw new NotImplementedException("The member IAsyncOperation<PdfDocument> PdfDocument.LoadFromFileAsync(IStorageFile file, string password) is not implemented in Uno.");
+		}
+
+		if (file is null)
+		{
+			throw new ArgumentNullException(nameof(file));
+		}
+
+		var localpath = file.Path;
+		var fileDescriptor = localpath.StartsWith('/')
+			? ParcelFileDescriptor.Open(new Java.IO.File(localpath), ParcelFileMode.ReadOnly)
+			: Android.App.Application.Context.ContentResolver.OpenFileDescriptor(Android.Net.Uri.Parse(localpath), "r");
+		var pdfRenderer = new PdfRenderer(fileDescriptor);
+		var pdfDocument = new PdfDocument(pdfRenderer);
+
+		return Task.FromResult(pdfDocument).AsAsyncOperation();
+	}
+
+	public static Foundation.IAsyncOperation<PdfDocument> LoadFromStreamAsync(Storage.Streams.IRandomAccessStream inputStream)
+	{
+		return LoadFromStreamAsync(inputStream, default);
+	}
+
+	public static Foundation.IAsyncOperation<PdfDocument> LoadFromStreamAsync(Storage.Streams.IRandomAccessStream inputStream, string password)
+	{
+		if (!string.IsNullOrWhiteSpace(password))
+		{
+			// password protected PDF files are not supported by PdfRenderer in Android
+			throw new NotImplementedException("The member IAsyncOperation<PdfDocument> PdfDocument.LoadFromStreamAsync(IRandomAccessStream inputStream, string password) is not implemented in Uno.");
+		}
+
+		var tcs = new TaskCompletionSource<PdfDocument>();
+		Task.Factory.StartNew(static arg =>
```

Comment:
Considering the low number of invocations of this particular method, using a non-static lambda will make the code a lot easier to read. Also, were you unable to use `AsyncOperation.FromTask`?

Reply:
I did not know `AsyncOperation.FromTask` :sweat_smile:.  Do I use it instead of `TaskCompletionSource`?

Reply:
It's best to use this yes, as it also provides cancellationtoken.